### PR TITLE
Fix opensearch grafana plugin at last working version

### PR DESCRIPTION
**Pull request by sjpb**
Without this PR the grafana plugin installed on a new deployment passes the GUI "test" button but can't actually talk to opensearch. Grafana returns a 500 error and the relevant tile in explore mode shows a JS trace containing `TypeError: pa[t] is undefined`.

Matt A worked out the last working version manually.

This also incorporates https://github.com/stackhpc/ansible-slurm-appliance/pull/293 to set the query type correctly. As noted there, this is because of changes to how querytype is handled by the plugin.

Note that:
- Idempotency was broken before this PR as [this difference](https://github.com/cloudalchemy/ansible-grafana/blob/b77b25607003c4cd9fd03091d03119d1dc8c2700/tasks/plugins.yml#L14) doesn't do what's intended - elements in `.files` are not plugin names but dicts including e.g. file stats.
- At this plugin version given here, the datasource still complains (in the edit DS GUI) about needing a version, but this does not prevent it working.

### CODE REVIEW

**Comment by mkarpiarz** [comment]
@sjpb: Could you please check if https://github.com/stackhpc/ansible-slurm-appliance/pull/293 fixes this problem?

Pinning this plugin to a specific version would still be useful to have but, in my opinion, should be handled by the collection.

**Comment by sjpb** [comment]
Cancelled CI as requires #294 to merge first before CI will complete

File: environments/common/inventory/group_vars/all/grafana.yml [hunk 1]
```diff
@@ -74,12 +75,15 @@ grafana_datasources:
...
+      # Have to set flavor and version, but ansible/roles/opensearch/templates/opensearch.yml.j2 fakes version for filebeat
+      # so need to set to fake version here:
+      version: '7.10.2'
+      flavor: elasticsearch
     editable: true
     # readOnly: false
 
 grafana_plugins:
-  - grafana-opensearch-datasource
+  - grafana-opensearch-datasource 2.8.1
```
Comment (m-bull) [inline environments/common/inventory/group_vars/all/grafana.yml:86]: Does it make sense to expose this as its own variable: `grafana_opensearch_datasource_version` or something?

**Review by m-bull — CHANGES_REQUESTED** [review]

File: environments/common/inventory/group_vars/all/grafana.yml (hunk 1 above)
Comment (sjpb) [reply environments/common/inventory/group_vars/all/grafana.yml:86]: Personally, I don't think so. It'd just more templating and vars in common, when really we want less there, without actually adding making it more customisable.

**Comment by sjpb** [comment]
@ af1c633492e3dd6c5cd8cfb2a21cdf50fe1a3ce9, confirmed slurm jobs dashboard works.

image

**Review by m-bull — APPROVED** [review]

**Pull request merged**